contracts/BatchExchange.sol

```diff
@@ -424,6 +427,41 @@ contract BatchExchange is EpochTokenLocker {
         return getEncodedUserOrdersPaginated(user, 0, uint16(-1));
     }
 
+    /** @dev View returning byte-encoded sell orders in paginated form
+      * @param previousPageUser address of last user received in the previous page (address(0) for first page)
+      * @param previousPageUserOffset the number of orders received for the last user on the previous page (0 for first page).
+      * @param pageSize uint determining the count of orders to be returned per page
+      * @return encoded bytes representing a page of orders ordered by (user, index)
+      */
+    function getEncodedUsersPaginated(address previousPageUser, uint16 previousPageUserOffset, uint16 pageSize)
+        public
+        view
+        returns (bytes memory elements)
+    {
+        if (allUsers.size() == 0) {
+            return elements;
+        }
+        uint16 currentOffset = previousPageUserOffset;
+        address currentUser = previousPageUser;
+        if (currentUser == address(0x0)) {
+            currentUser = allUsers.first();
+        }
+        while (elements.length / ENCODED_AUCTION_ELEMENT_WIDTH < pageSize) {
+            elements = elements.concat(
+                getEncodedUserOrdersPaginated(
```

Comment:
It is worth checking whether or not the order is valid before reading it? Maybe `encodeAuctionElement` can first read what it needs to determine order validity and then read the rest of the storage if the order is valid. Returning empty bytes if the order is invalid and encoded bytes if it is.

Reply:
Yes. Like filtering out expired orders! Maybe even filtering out orders that are > 99% filled too!